src/server/journal/streamer.cc

```diff
@@ -244,14 +249,21 @@ RestoreStreamer::~RestoreStreamer() {
 void RestoreStreamer::Cancel() {
   auto sver = snapshot_version_;
   snapshot_version_ = 0;  // to prevent double cancel in another fiber
+  fiber_cancelled_ = true;
   if (sver != 0) {
-    fiber_cancelled_ = true;
     db_slice_->UnregisterOnChange(sver);
     JournalStreamer::Cancel();
   }
 }
 
 bool RestoreStreamer::ShouldWrite(const journal::JournalItem& item) const {
+  if (item.cmd == "FLUSHALL" || item.cmd == "FLUSHDB") {
```

Comment:
it's not only about flushALL and flushDB. We have several more global commands. You can find a list in TransactionData::IsGlobalCmd()

Reply:
We have special handling for `FLUSH*` commands in replication as well, because these are the only commands which actually drop a db table.
Generally speaking, global commands are not supposed to be sent via migration, and so it wouldn't make sense to break a connection when we get them (it's even worse, as `DFLYCLUSTER` is one of those commands, so an attempt to even query the status of a migration will kill it..)